### ISSUE TITLE
Game was broken when unpacking updates

### ISSUE THREAD

**Issue by labo-China** [an-anime-team/an-anime-game-launcher#82]
As the title, my game is broken when unpacking updates. The main executable file "GenshinImpact.exe", "mhyprot3.sys" etc. were gone and I am unable to install updates because the program reports that "Failed to find game diff". The folder "GenshinImpact_Data" seems to be OK.

My game was on a EXFAT partition and it was linked into my wine prefix where was written in config.json. So the situation is like the graph below:
Original Game(on EXFAT and broken) ===> folder in wine prefix <=== your launcher
When I attempted to install updates, the download process is fine. But it stopped with a prompt "Operation not premitted"(translated from Chinese). The progress bar stopped at around 6kib. And I also find a file called "deletefiles.txt" in the game directory.

In one word, a error occured when unpacking updates and broke my game installation. It might relate to EXFAT because the "Operation not premitted" prompt also occured when I try to install games by steam. However, I have runned fsck and it turns out that the partition is fine and it works well with daily use(coding with pycharm, using git etc.)

I wonder if there is a way to patch the game manually without reinstalling the whole game. Because my free disk space(~15 GB) can barely afford it. BTW, I am a student preparing for a exam so I maybe cannot reply in time. I am looking forward to any help and answers!

**Comment by krypt0nn** [comment]
Maybe also be related to use of symlinks. You can set all paths in config file, don't use symlinks

About manual patching:

https://notabug.org/Krock/dawn/src/master/350
https://notabug.org/Krock/dawn/src/master#applying-the-workaround

**Comment by labo-China** [comment]
Thanks for your help and sorry for my ambiguous expression. I actually want to just update the game to ver 3.5. I have downloaded the ~6GB zip file. How can I apply it to my game? Or do I need some modifications to do?

**Comment by krypt0nn** [comment]
Just let launcher do its job and press "update" button. It won't redownload archive (so continue if needed), and unpack it once it fully downloaded

**Comment by labo-China** [comment]
Good suggestions! But I have some problems now:
1. The official launcher doesn't detect my game installation(even the installation was not broken)
2. This launcher seems unable to update successfully as it prompts “Failed to find game diff” and the update button become gray and shows “...”
So I think I should fix the game and try to update by launchers, or I need to update manually.
What should I do?

**Comment by labo-China** [comment]
I ran the script in the launcher dir to apply update package mannualy and solved the problem. So the game is good now. 
When I ran the script, it still prompts something like "Operation not premitted", but everything is OK. Maybe the problem is that EXFAT does not support modifing some filesystem attributes (access time etc.). It causes softwares like hpatchz or something else raise a warning when writing files and therefore return a non-zero exit code and the launcher treats the code as a error and exit. Is that a expected behavior? If it is, I will close this issue.
Thanks for any help and sorry for bothering you.
